### DISCUSSION TITLE
Datagrid Multiselect

### DISCUSSION THREAD

**Discussion by kyotr** [Megabit/Blazorise#5501] [Q&A] [answered]
Hello,

i would like to multi select rows only by using DataGridMultiSelectColumn column.  clicking on the row should select only one row; in case of multiselect i will use DataGridMultiSelectColumn

STEP 1:
![image](https://github.com/Megabit/Blazorise/assets/22255109/73ee4cad-7741-491b-a674-dfa04631917b)

STEP2:
when i click on constructions row it should deselect all previous rows and select only the targeted row "CONSTURCTIONS"  

![image](https://github.com/Megabit/Blazorise/assets/22255109/6904ba2e-9f49-41d0-9a44-f9f6ab569f33)

any ideas ? 

Regards.

**Comment by stsrki** [comment]
@David-Moreira can you please help?

**Reply by David-Moreira** [reply, marked as answer]
Unfortunately we do not support explicitly changing the behaviour. This is because the multi select can both work with or without checkboxes.

You could however try 
- overriding the behavior by listening to the SelectedRowChanged or RowClicked event and clearing the SelectedRows accordingly?
- override RowSelectable to return false on row click which will disable any internal selection logic, and select the row yourself

So hopefully there should be a few ways you can override the behavior to yield the expected result.

**Comment by kyotr** [comment]
Hello @David-Moreira 

Thank you for your answer, adding this line helped me 

` RowClicked="row => selectedRows.Clear( )"` 

Many thanks
